rest-service/manager_rest/cluster_status_manager.py

```diff
@@ -39,6 +47,83 @@
 CLUSTER_STATUS_PATH = '/opt/manager/cluster_statuses'
 
 
+# region Syncthing Status Helpers
+
+
+def _last_manager_in_cluster():
+    storage_manager = get_storage_manager()
+    managers = storage_manager.list(models.Manager,
+                                    sort={'last_seen': 'desc'})
+    active_managers = 0
+    for manager in managers:
+        # Probably new manager, first status report is yet to arrive
+        if manager.status_report_frequency is None:
+            active_managers += 1
+        else:
+            # The manager is considered active, if the time passed since
+            # it's last_seen is maximum twice the frequency interval
+            # (Nyquist sampling theorem)
+            interval = manager.status_report_frequency * 2
+            min_last_seen = datetime.utcnow() - timedelta(seconds=interval)
+
+            if parse_datetime_string(manager.last_seen) > min_last_seen:
+                active_managers += 1
+        if active_managers > 1:
+            return False
+    return True
+
+
+def _other_device_was_seen(syncthing_config, device_stats):
+    # Add 1 second to the interval for avoiding false negative
+    interval = syncthing_config['options']['reconnectionIntervalS'] + 1
+    min_last_seen = datetime.utcnow() - timedelta(seconds=interval)
+
+    for device_id, stats in device_stats.items():
+        last_seen = parse_datetime_string(stats['lastSeen'])
+
+        # Syncthing is valid when at least one device was seen recently
+        if last_seen > min_last_seen:
+            return True
+    return False
+
+
+def _is_syncthing_valid(syncthing_config, device_stats):
```

Comment:
_is_syncthing_valid => _is_manager_cluster_structure_valid

Or something like that, because syncthing is the way we check the manager cluster structure

Reply:
I disagree, the intention of the check is to validate syncthing is working properly and sync the file system of all the active managers.